pkg/genericapiserver/genericapiserver.go

```diff
@@ -385,6 +354,11 @@ func (s *GenericAPIServer) Run(options *options.ServerRunOptions) {
 	<-insecureStartedCh
 	s.RunPostStartHooks(PostStartHookContext{})
 
+	// err == systemd.SdNotifyNoSocket when not running on a systemd system
+	if err := systemd.SdNotify("READY=1\n"); err != nil && err != systemd.SdNotifyNoSocket {
```

Comment:
@deads2k It's the opposite, we notify systemd that we're ready so that other services that depend on the api server don't try to start until the api is available.